src/ray/raylet/worker_pool.cc

```diff
@@ -571,6 +593,15 @@ void WorkerPool::HandleJobFinished(const JobID &job_id) {
   // Currently we don't erase the job from `all_jobs_` , as a workaround for
   // https://github.com/ray-project/ray/issues/11437.
   // unfinished_jobs_.erase(job_id);
+  auto job_config = GetJobConfig(job_id);
+  RAY_CHECK(job_config);
+  if (job_config->has_runtime_env_info() &&
+      job_config->runtime_env_info().runtime_env_eager_install()) {
```

Comment:
Can you comment why we have this eager_install check in the if statement? Hard to understand the reasoning without that if you are not familiar with runtime env

Reply:
Yep, I can add a comment. We just add uri reference when runtime env install happens really.